src/test/pmem_reorder_simple/pmem_reorder_simple.c

```diff
@@ -0,0 +1,140 @@
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * pmem_reorder_simple.c -- a simple unit test for store reordering
+ *
+ * usage: pmem_reorder_simple file g|b|c
+ * g - write data in a consistent manner
+ * b - write data in a possibly inconsistent manner
+ * c - check data consistency
+ *
+ */
+
+#include "unittest.h"
+#include "util.h"
+#include "valgrind_internal.h"
+
+/*
+ * The struct three_field is inconsistent if flag is set and the fields have
+ * different values.
+ */
+struct three_field {
+	int first_field;
+	int second_field;
+	int third_field;
+	int flag;
+};
+
+/*
+ * write_consistent -- (internal) write data in a consistent manner
+ */
+static void
+write_consistent(struct three_field *structp)
+{
+	structp->first_field = 1;
+	structp->second_field = 1;
+	structp->third_field = 1;
+	pmem_persist(&structp->first_field, sizeof(int) * 3);
+	structp->flag = 1;
+	pmem_persist(&structp->flag, sizeof(structp->flag));
+}
+
+/*
+ * write_inconsistent -- (internal) write data in an inconsistent manner.
+ */
+static void
+write_inconsistent(struct three_field *structp)
+{
+	structp->first_field = 1;
+	structp->second_field = 1;
+	structp->third_field = 1;
+	structp->flag = 1;
+	pmem_persist(structp, sizeof(*structp));
+}
+
+/*
+ * check_consistency -- (internal) check struct three_field consistency
+ */
+static int
+check_consistency(struct three_field *structp)
+{
+	int consistent = 0;
+	if (structp->flag)
+		consistent = (structp->first_field != structp->second_field) ||
+			(structp->first_field != structp->third_field);
+	return consistent;
+}
+
+int
+main(int argc, char *argv[])
+{
+	START(argc, argv, "pmem_reorder_simple");
+
+	util_init();
+
+	if ((argc > 4) || (strchr("gbc", argv[1][0]) == 0))
```

Comment:
IMHO, validation of input parameters could be a little more restrictive:
-You could additionally confirm that argv[1][1] is equal '\0' - currently it is possible to pass as first argument "c" and for example "cgbcgb" - which will be interpreted as "c". Only "c" should be valid, not "cgbcgb"
-argc could check here lower limit ( it shouldn't be less than 3 )
-In Your solution (TEST0) You pass 3 arguments (argc=4), which is inconsistent with usage ( it described only 2 parameters **g|b|c** and **file** ) - You should update usage section or TEST0
Additionally IMHO cosmetics: strchr function should compare to NULL instead of 0. strchr function returns pointer. I know it could be compare to 0, but compare to NULL gives You clear information, about type which strchr returns, eventually use !strchr("gbc", argv[1][0]). Compare here to 0 is misleading.